### PR TITLE
tools/expat: update to 2.4.6

### PR DESCRIPTION
**Pull request by neheb**
Switched to CMake for faster compilation and greater parallel
friendliness.

Added CMake options from the packages feed.

This release fixes various CVEs.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

ping @BKPepe

### CODE REVIEW

**Comment by BKPepe** [comment]
> This release fixes various CVEs.

Oh right, this is important. Similar PR https://github.com/openwrt/packages/pull/17931 for packages was sent by @paper42. Then I cherry-picked it to OpenWrt 19.07. What I am trying to say is that this update (w/o cmake changes, etc.) should be done as well for OpenWrt 21.02 and OpenWrt 19.07. I can take care of it and send pull requests, if requested.

> Switched to CMake for faster compilation and greater parallel

I will push such change to Turris OS to test it and let you know about it. But that's not going to be done before the afternoon.

File: tools/expat/Makefile [hunk 1]
```diff
@@ -9,21 +9,25 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=expat
 PKG_CPE_ID:=cpe:/a:libexpat:expat
-PKG_VERSION:=2.2.10
+PKG_VERSION:=2.4.6
```
Comment (BKPepe) [inline tools/expat/Makefile:12]: Btw, I noticed mismatch in the commit subject and with this version

``2.4.1`` vs ``2.4.6``

Comment (neheb) [reply tools/expat/Makefile:12]: Fixed.

**Comment by neheb** [comment]
Why without cmake changes?

File: tools/expat/Makefile (hunk 1 above)
Comment (BKPepe) [reply tools/expat/Makefile:12]: :+1:

**Comment by BKPepe** [comment]
It is a question, if you want to introduce such changes there if it should be a stable branch. But if it is going to be tested properly and as well with time delay it could be backported there, too. But of course, it should work and not cause any unwanted regressions.

**Comment by neheb** [comment]
As far as I can tell, tools/expat is only used for toolchain/gdb. Wonder why. Would be interesting to try removing tools/expat and see what breaks.

**Comment by neheb** [comment]
Alright, my conclusion is that tools/expat can be removed. I installed both a lightweight Alpine Linux container and a lightweight debian container. Both include expat. The former includes it when installing git.

However, it might be needed under macOS.

@svlobanov can you test https://github.com/neheb/openwrt/commit/e4d8687f12d9570dcb03635fe2494adb4de4b28f and see if this is the case?

**Comment by hauke** [comment]
Thank you for your patch, I applied it to master in 4e13229dd19e90975af9a51b4bc888477fbfad5e.

**Pull request closed without merging**